exchanges/binance/binance_wrapper.go

```diff
@@ -122,6 +125,22 @@ func (b *Binance) SetDefaults() {
 	b.WebsocketOrderbookBufferLimit = exchange.DefaultWebsocketOrderbookBufferLimit
 }
 
+// RateLimit implements the request.Limiter interface
+type RateLimit struct {
```

Comment:
Yeah I see what you're saying but its the matching and readability for the functionality. And some require an overarching global rate limit so defining that on the request package side will be unfortunate. But when you get back in we can go over some examples to stream line it up.

Reply:
From looking into it further, definitely understand you when it comes to the `Limit` function. Per exchange, you definitely need individual implementations. I was hoping for a way to minimise code, struct and property name reuse. But I think it would be too much and I'm happy with what is here